src/Route.php

```diff
@@ -297,31 +282,29 @@ public function __debugInfo()
             'defaults' => $this->defaults,
             'override' => $this->override,
             'actionAdded' => $this->actionAdded,
-            'middlewareDefinitions' => $this->middlewareDefinitions,
-            'disabledMiddlewareDefinitions' => $this->disabledMiddlewareDefinitions,
-            'middlewareDispatcher' => $this->dispatcher,
+            'middlewares' => $this->middlewares,
+            'disabledMiddlewares' => $this->disabledMiddlewares,
+            'enabledMiddlewares' => $this->getEnabledMiddlewares(),
         ];
     }
 
-    private function getDispatcherWithMiddlewares(): MiddlewareDispatcher
+    /**
+     * @return array[]|callable[]|string[]
+     * @psalm-return list<array|callable|string>
+     */
+    private function getEnabledMiddlewares(): array
     {
-        if ($this->dispatcher === null) {
-            throw new RuntimeException(sprintf('There is no dispatcher in the route %s.', $this->getData('name')));
+        if ($this->enabledMiddlewares !== null) {
+            return $this->enabledMiddlewares;
         }
 
-        // Don't add middlewares to dispatcher if we did it earlier.
-        // This improves performance in event-loop applications.
-        if ($this->dispatcher->hasMiddlewares()) {
-            return $this->dispatcher;
-        }
-
-        /** @var mixed $definition */
-        foreach ($this->middlewareDefinitions as $index => $definition) {
-            if (in_array($definition, $this->disabledMiddlewareDefinitions, true)) {
-                unset($this->middlewareDefinitions[$index]);
-            }
-        }
+        $this->enabledMiddlewares = array_values(
+            array_filter(
```

Comment:
`array_filter` function is faster than `foreach`?

Reply:
Seems, foreach faster (I'm search in web, not make own bench). Returned `foreach`.